cypresse2e/cypress/integration/e2eTests/helpBoard/helpBoard.spec.js

```diff
@@ -21,12 +24,63 @@ describe("FightPandemics Help Board Page for unauthorized user", () => {
       cy.validateCorrectScreenIsOpen("auth/login");
     });
 
-    it.skip("Help Board is empty - Unauthorized user is redirected to SignIn page when clicking Add Post link", () => {
+    it("keyword search container is visible and clickable", () => {
+      helpBoard.getSearchContainerDiv().should("be.visible").click();
+    });
+    it(" clicking on search container, searchbox for typing opens", () => {
+      searchInputForPosts();
+    });
+
+    it("cancel button 'x' in search is visible and clickable", () => {
+      searchInputForPosts().type(inputValues.searchByRandomText + "{enter}");
+      helpBoard
+        .getSearchCancelButton()
+        .should("be.visible")
+        .click({ force: true });
+    });
+    it("Unauthorized user can search posts by typing name of an user", () => {
+      searchInputForPosts().type(inputValues.searchByName + "{enter}");
+      helpBoard.getAuthorTitle().should("be.visible").contains(authorTitleText);
+    });
+
+    it("Unauthorized user if type single character for search", () => {
+      searchInputForPosts().type(
+        inputValues.searchBySingleCharacter + "{enter}",
+      );
+      helpBoard
+        .getErrorMessage()
+        .should("be.visible")
+        .contains("small", errorMessages.errorMessageTextInSearch);
+    });
+    it("No post appear if unauthorized user search posts by typing misspelled name", () => {
+      searchInputForPosts().type(
+        inputValues.searchByMisspelledText + "{enter}",
+      );
+      helpBoard.getAddPostLinkElement();
+      helpBoard
+        .getNoPostsAvailableTextElement()
+        .contains(errorMessages.noPostsAvailableText);
+    });
+
+    it("Help Board is empty - Unauthorized user is redirected to SignIn page when clicking Add Post link", () => {
       //there is no post on the Help Board
-      helpBoard.getNoPostsAvailableTextElement().contains(noPostsAvailableText);
+      searchInputForPosts().type(
+        inputValues.searchByMisspelledText + "{enter}",
+      );
+      helpBoard
+        .getNoPostsAvailableTextElement()
+        .contains(errorMessages.noPostsAvailableText);
+      helpBoard.getAddPostLinkElement();
```

Comment:
I don't think we need to include line 73 `helpBoard.getAddPostLinkElement();` since this is already tested below in lines 74-76.

Reply:
Agree

Reply:
I agree with the previous suggestions